### DISCUSSION TITLE
How to use templ.ComponentScript along with @click or x-on:click (of alpinejs)

### DISCUSSION THREAD

**Discussion by navamedha** [a-h/templ#567] [Q&A]
Currently, I can use `script onTableTHClick(colName string) {}` along with `onclick` attribute.

But when I use the same along with `@click` or `x-on:click` of alpinejs, templ is reporting the following error.

`cannot use onTableTHClick(colName) (value of type templ.ComponentScript) as string value in argument to templ.EscapeStringcompiler[IncompatibleAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#IncompatibleAssign)`

How to use script components along with attributes other than `onclick` ?

<img width="933" alt="Screenshot 2024-02-28 at 12 17 22" src="https://github.com/a-h/templ/assets/105703058/a9be4477-b3e3-410d-8b93-af01124ce842">

**Comment by ikaio** [comment]
I think `.Call` is what you looking for: https://pkg.go.dev/github.com/a-h/templ#ComponentScript
```go
@click={ onTableTHClick(colName).Call } // not tested
```
